biocrnpyler/components/membrane.py

```diff
@@ -3,45 +3,41 @@
 #  See LICENSE file in the project root directory for details.
 
 from typing import List, Union
+import warnings
 
 from ..core.component import Component
 from ..core.reaction import Reaction
 from ..core.species import Complex, Species
+from ..core.compartment import Compartment
 
 class DiffusibleMolecule(Component):
     """A class to represent transmembrane proteins or integral membrane proteins.
     This membrane class is to classify a membrane channel that will intergrate into the membrane.
     By default, a DiffusibleMolecule uses a mechanism called "diffusion".
     """
-    def __init__(self, substrate: Union[Species, str, Component], 
-                 internal_compartment:str='Internal', external_compartment:str='External',
-                 cell:Union[str, int]=None, attributes=None, **keywords):       
+    def __init__(self, substrate:Union[Species, str, Component], 
+                 internal_compartment:Union[str, Compartment]='Internal', 
+                 external_compartment:Union[str, Compartment]='External',
+                 attributes=None, **keywords):       
         """Initialize a DiffusibleMolecule object to store molecule/substance related information.
         :param substrate: name of the diffusible substrate, reference to an Species or Component
         :param internal_compartment: name of internal compartment 
         :param external_compartment: name of external compartment 
-        :param cell: indicates the cell, identifier can be name (str) or number (int)
         :param attributes: Species attribute, passed to Component
         :param keywords: pass into the parent's (Component) initializer
         """
-
-        # Additional information on the identity of the specific cell/vesicle (if needed).
-        if cell is not None:
-            if type(cell) is str:
-                internal_compartment=internal_compartment+'_'+cell
-                external_compartment=external_compartment
-            else:
-                internal_compartment=internal_compartment+'_'+str(cell)
-                external_compartment=external_compartment
+        # Creates compartment object if compartment is a str
+        if type(internal_compartment) is str:
+            internal_compartment = Compartment(name=internal_compartment)
+        if type(external_compartment) is str:
+            external_compartment = Compartment(name=external_compartment)
+        
         # Substrate
         self.substrate = self.set_species(substrate, compartment=internal_compartment)
         self.product = self.set_species(substrate, compartment=external_compartment)
         
         # Name the component
-        if cell is not None:
-            name = self.substrate.name + '_' + str(cell)
-        else:
-            name = self.substrate.name
+        name = self.substrate.compartment.name
```

Comment:
Shouldn't this be `self.substrate.name`?  It seems like we want to name the component after the substrate species and not after the compartment that it is in (consistent with what was done before).

Reply:
Yes, it looks like it should be `substrate.name`. @zjuradoq are we missing something here?

Reply:
@murrayrm please refer to above, I was not able to type faster than your comment about failing the unit test.